website/docs/docs/build/udfs.md

```diff
@@ -0,0 +1,152 @@
+---
+title: "User-defined functions"
+description: "Learn how to add user-defined functions (UDFs) to your dbt projects."
+id: "udfs"
+---
+
+# User-defined functions <Lifecycle status="beta" />
+
+<!--Confirm how users can get this feature-->
+:::info Beta feature 
+If you are interested in getting access to this feature, please contact us. 
+:::
+
+User-defined functions (UDFs) enable users to define and register custom functions within the warehouse. Like [macros](/docs/build/jinja-macros), UDFs enable reuse of code; however, unlike macros, you can define UDFs in languages other than SQL (for example, Python, Java, Scala) and you can use them in queries outside the <Constant name="dbt_platform" />.
+
+Note that only basic SQL UDFs are currently supported in dbt.<!--Is this correct?-->
+
+dbt creates, updates, and renames UDFs as part of DAG execution. The UDF file is created before building the model that references it.
+
+## Supported adapters
+
+UDFs are supported in the following adapters:
+
+- BigQuery
+- Snowflake
+- Redshift
+
+## Defining UDFs in dbt
+
+To define UDFs in dbt, refer to the following steps:
+
+1. Create a SQL file under the `functions` directory. For example:
+
+    <File name='functions/is_positive_int.sql'>
+
+    ```sql
+    {{ config(
+        database='udf_db'
+    ) }}
+
+    REGEXP_CONTAINS(a_string, r'^[0-9]+$')
+    ```
+
+    </File>
+
+2. Define your argument, output types, properties, and configs in a corresponding YAML file. For example:
+
+
+    <File name='functions/schema.yml'>
+
+    ```yml
+    functions:
+      - name: is_positive_int # required
+        description: My UDF that determines if a string represents a positive (+) integer # required
+        config:
+            schema: udf_schema
+            database: udf_db
+        arguments: 
+            - name: a_string
+              type: string
+              description: The string that I want to check if it's representing a positive integer (like "10") 
+        return_type: # required
+            type: boolean # required
```

Comment:
This is my fault in. I originally implemented them incorrectly, but they are now fixed
* `arguments > type` should actually be `arguments > data_type`
* `return_type` should be `returns`
* `return_type > type` should be `returns > data_type`